### PR TITLE
Pretty big rework

### PR DESCRIPTION
**Pull request by brandonferens**

### CODE REVIEW

File: src/Action.php [hunk 1]
```diff
@@ -3,68 +3,86 @@
...
      * @throws Throwable
      *
      * @return mixed|void
      */
-    public function act(Actionable $action)
+    public function act(string $action)
     {
+        $this->arguments = array_slice(func_get_args(), 1);
+
         return $this->handle($action);
```
Comment (babacarcissedia) [inline src/Action.php]: some magic here ✨

File: src/ActionsServiceProvider.php [hunk 2]
```diff
@@ -15,15 +19,88 @@ class ActionsServiceProvider extends ServiceProvider
      */
     public $bindings = [
         'actions' => Action::class,
-        Actionable::class => Action::class,
+        Action::class => Action::class,
```
Comment (babacarcissedia) [inline src/ActionsServiceProvider.php:22]: no more binding with the interface ? Curious about this. Is there any other reason why except now being able to have container resolve classes for us ?

File: src/ActionsServiceProvider.php [hunk 3]
```diff
@@ -15,15 +19,88 @@ class ActionsServiceProvider extends ServiceProvider
...
+        if ($paths->isEmpty()) {
+            return;
+        }
+
+        $namespace = $this->app->getNamespace();
+
+        foreach ((new Finder())->in($paths->toArray())->files() as $action) {
+            $action = $namespace . str_replace(
+                ['/', '.php'],
+                ['\\', ''],
```
Comment (babacarcissedia) [inline src/ActionsServiceProvider.php:85]: didn't know we can do that 🤯

File: src/Action.php [hunk 4]
```diff
@@ -3,68 +3,86 @@
...
+     * @param string $action
      *
      * @throws Throwable
      *
      * @return mixed|void
      */
-    public function act(Actionable $action)
+    public function act(string $action)
     {
+        $this->arguments = array_slice(func_get_args(), 1);
```
Comment (alexandersix) [inline src/Action.php]: Could we use the `...` syntax here (and the other actX methods)? Could be handy for LSPs that show what parameters can be passed into a method.

For example:
```suggestion
    public function act(string $action, ...$arguments)
    {
        $this->arguments = $arguments;
```

**Review by alexandersix — CHANGES_REQUESTED** [review]
Looks good–just one thing to consider with the `func_get_args`.

**Review by babacarcissedia — APPROVED** [review]
Looking good to me

File: src/ActionsServiceProvider.php (hunk 3 above)
Comment (brandonferens) [reply src/ActionsServiceProvider.php:85]: Doing the array thing is super handy during string replacements

File: src/Action.php (hunk 4 above)
Comment (brandonferens) [reply src/Action.php]: You are absolutely right

**Review by alexandersix — APPROVED** [review]

**Pull request merged**